GitUI/UserControls/ToolStripEx.cs

```diff
@@ -18,6 +19,64 @@ public ToolStripEx()
             _gripButton = propGrip.GetValue(this) as ToolStripButton;
         }
 
+        protected override void OnItemAdded(ToolStripItemEventArgs e)
+        {
+            base.OnItemAdded(e);
+
+            if (e.Item is ToolStripDropDownItem item)
+            {
+                item.DropDownOpening += SplitButton_DropDownOpening;
+                item.DropDownClosed += SplitButton_DropDownClosed;
+            }
+        }
+
+        protected override void OnItemRemoved(ToolStripItemEventArgs e)
+        {
+            if (e.Item is ToolStripDropDownItem item)
+            {
+                item.DropDownOpening -= SplitButton_DropDownOpening;
+                item.DropDownClosed -= SplitButton_DropDownClosed;
+            }
+        }
+
+        [DllImport("user32.dll")]
+        private static extern int SendMessage(IntPtr hwnd, int wmsg, bool wparam, int lparam);
+
+        private const int WM_SETREDRAW = 11;
+
+        private static void SuspendDrawing(Control control)
+        {
+            if (control != null)
+            {
+                SendMessage(control.Handle, WM_SETREDRAW, false, 0);
+            }
+        }
+
+        private static void ResumeDrawing(Control control)
+        {
+            if (control != null)
+            {
+                SendMessage(control.Handle, WM_SETREDRAW, true, 0);
+                control.Refresh();
+            }
+        }
+
+        private void SplitButton_DropDownOpening(object? sender, EventArgs e)
+        {
+            if (sender is ToolStripDropDownItem item)
```

Comment:
```suggestion
            if (sender is ToolStripDropDownItem item && item.Owner is not null)
```

ditto below

Reply:
Let's assume we keep `SuspendDrawing()` and `ResumeDrawing()` not inlined, and since the first thing they do is to check for `null`, would this change still be necessary ?

Reply:
> Would this change still be necessary?

only if the functions are merged (I tend to keep them separate but YAGNI.)

Reply:
I added this check since methods have been inlined.